python/lsst/dax/apdb/cassandra/config.py

```diff
@@ -0,0 +1,218 @@
+    )
+
+    time_partition_tables: bool = Field(
+        default=False,
+        description="Use per-partition tables for sources instead of partitioning by time",
+    )
+
+    time_partition_days: int = Field(
+        default=30,
+        description=(
+            "Time partitioning granularity in days, this value must not be changed after database is "
+            "initialized"
+        ),
+    )
+
+    time_partition_start: str = Field(
+        default="2018-12-01T00:00:00",
+        description=(
+            "Starting time for per-partition tables, in yyyy-mm-ddThh:mm:ss format, in TAI. "
+            "This is used only when time_partition_tables is True."
+        ),
+    )
+
+    time_partition_end: str = Field(
+        default="2030-01-01T00:00:00",
+        description=(
+            "Ending time for per-partition tables, in yyyy-mm-ddThh:mm:ss format, in TAI. "
+            "This is used only when time_partition_tables is True."
+        ),
+    )
+
+    query_per_time_part: bool = Field(
+        default=False,
+        description=(
+            "If True then build separate query for each time partition, otherwise build one single query. "
+            "This is only used when time_partition_tables is False in schema config."
+        ),
+    )
+
+    query_per_spatial_part: bool = Field(
+        default=False,
+        description="If True then build one query per spatial partition, otherwise build single query.",
+    )
+
+    @field_validator("part_pixelization")
+    @classmethod
+    def check_pixelization(cls, v: str) -> str:
+        allowed = {"htm", "q3c", "mq3c"}
+        if v not in allowed:
+            raise ValueError(f"Unexpected value for part_pixelization: {v}, allowed values: {allowed}")
+        return v
+
+
+class ApdbCassandraConfig(ApdbConfig):
+    """Configuration class for Cassandra-based APDB implementation."""
+
+    _implementation_type: ClassVar[str] = "cassandra"
+
+    contact_points: list[str] = Field(
+        default=["127.0.0.1"],
+        description="The list of contact points to try connecting for cluster discovery.",
+    )
+
+    keyspace: str = Field(
+        default="apdb",
+        description="Keyspace name for APDB tables.",
+    )
+
+    connection_config: ApdbCassandraConnectionConfig = Field(
+        default_factory=ApdbCassandraConnectionConfig,
+        description="Database connection configuration",
+    )
+
+    partitioning: ApdbCassandraPartitioningConfig = Field(
+        default_factory=ApdbCassandraPartitioningConfig,
+        description="Configuration for partitioning.",
+    )
+
+    dia_object_columns: list[str] = Field(
+        default=[],
+        description="List of columns to read from DiaObject[Last], by default read all columns.",
+    )
+
+    prefix: str = Field(
+        default="",
+        description="Prefix to add to table names.",
+    )
+
+    ra_dec_columns: list[str] = Field(
```

Comment:
list or tuple? `tuple[str, str]` seems better since you know it has to be two values. A list would need a validation step wouldn't it?

Reply:
There is indeed a validation method for this. I'll switch to tuple.

Reply:
It's a bit more complicated, as I also need to parse JSON-originated dict and pydantic does not like JSON list in strict mode. I had to keep validation method to convert that list to tuple.